### DISCUSSION TITLE
Allow popping or updating the last line in `ui.log`

### DISCUSSION THREAD

**Discussion by virginiaWu** [zauberzeug/nicegui#2003] [Ideas / Feature Requests]
The Log View Content, like:

This is the first time input.
This is the second time input. This is the thrid time input.
This is the fourth time input.

**Comment by falkoschindler** [comment]
Sorry, I don't quite understand your question. Can you explain it with a minimal code example?

**Reply by virginiaWu** [reply]
OK.
```py
res_output = ui.log().style('margin-left: 10%; min-height: 430px; width: 85%;')
res_output.push("Start do power cycle")
s_time = time.time()
while time.time() - s_time < 30:
   res_output.push("..")    #  want to all "........." string in the one line, not ".." in one line, ".." in another line 
res_output.push("Power cycle done.")
```

**Reply by falkoschindler** [reply]
I see. Unfortunately `ui.log` is completely based on lines separated by `\n` and it's not possible to change lines that already have been pushed. As a workaround you can repeatedly `clear()` and `push()` the line. Otherwise we would need to extend `ui.log` to allow popping or updating the last line.

**Reply by virginiaWu** [reply]
I got it. Do you have the plan to extend the ui.log to allow popping or updating the last line?

**Reply by falkoschindler** [reply]
Not yet. But I'll convert this discussion into a feature request.

**Comment by natankeddem** [comment]
Depending on your use case you might find xterm useful. You should be able to interact with it just like a terminal although I haven't exercised it much with advanced inputs. This is a bit of a verbose example:
https://github.com/zauberzeug/nicegui/discussions/1846

Also, implemented slightly differently in my personal project:
https://github.com/natankeddem/bale/blob/ffbc9b71c0a1c6652dcfc94da26def067bd432c0/bale/interfaces/cli.py#L18
https://github.com/natankeddem/bale/blob/main/static/terminal.js
